cdap-unit-test/src/main/java/co/cask/cdap/test/internal/DefaultStreamManager.java

```diff
@@ -81,7 +83,8 @@ public DefaultStreamManager(StreamHandler streamHandler, StreamFetchHandler stre
   @Override
   public void createStream() throws IOException {
     String path = String.format("/v3/namespaces/%s/streams/%s", streamId.getNamespaceId(), streamId.getId());
-    HttpRequest httpRequest = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.PUT, path);
+    FullHttpRequest httpRequest = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.PUT, path);
+    HttpUtil.setContentLength(httpRequest, 0);
```

Comment:
we can't just avoid setting the content length?

Reply:
Maybe.

Reply:
For this one, better have the content length set, since the request is actually not going through the netty stack, but instead used to call the handler method directly (that's what the unit-test framework does).

For other places that has set content length to 0, those can be removed.